src/SprykerEco/Zed/Payone/Business/Payment/Hook/CheckoutPostSaveHookExecutor.php

```diff
@@ -0,0 +1,133 @@
+use SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface;
+
+class CheckoutPostSaveHookExecutor implements CheckoutPostSaveHookExecutorInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface
+     */
+    protected $queryContainer;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader
+     */
+    protected $paymentMapperReader;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\DataMapper\PayoneRequestProductDataMapperInterface
+     */
+    protected $payoneRequestProductDataMapper;
+
+    /**
+     * @var \SprykerEco\Zed\Payone\Business\Payment\MethodSender\PayoneBaseAuthorizeSenderInterface
+     */
+    protected $baseAuthorizeSender;
+
+    /**
+     * @param \SprykerEco\Zed\Payone\Persistence\PayoneQueryContainerInterface $queryContainer
+     * @param \SprykerEco\Zed\Payone\Business\Payment\PaymentMapperReader $paymentMapperReader
+     * @param \SprykerEco\Zed\Payone\Business\Payment\DataMapper\PayoneRequestProductDataMapperInterface $payoneRequestProductDataMapper
+     * @param \SprykerEco\Zed\Payone\Business\Payment\MethodSender\PayoneBaseAuthorizeSenderInterface $baseAuthorizeSender
+     */
+    public function __construct(
+        PayoneQueryContainerInterface $queryContainer,
+        PaymentMapperReader $paymentMapperReader,
+        PayoneRequestProductDataMapperInterface $payoneRequestProductDataMapper,
+        PayoneBaseAuthorizeSenderInterface $baseAuthorizeSender
+    ) {
+        $this->queryContainer = $queryContainer;
+        $this->paymentMapperReader = $paymentMapperReader;
+        $this->payoneRequestProductDataMapper = $payoneRequestProductDataMapper;
+        $this->baseAuthorizeSender = $baseAuthorizeSender;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponse
+     *
+     * @return \Generated\Shared\Transfer\CheckoutResponseTransfer
+     */
+    public function executeCheckoutPostSaveHook(
+        QuoteTransfer $quoteTransfer,
+        CheckoutResponseTransfer $checkoutResponse
+    ): CheckoutResponseTransfer {
+        if ($quoteTransfer->getPayment()->getPaymentProvider() !== PayoneConfig::PROVIDER_NAME) {
+            return $checkoutResponse;
+        }
+
+        $idSalesOrder = $checkoutResponse->getSaveOrder()->getIdSalesOrder();
+        $paymentEntity = $this->queryContainer->createPaymentById($idSalesOrder)->findOne();
+
+        $paymentMethodMapper = $this->paymentMapperReader->getRegisteredPaymentMethodMapper($paymentEntity->getPaymentMethod());
+        $requestContainer = $this->getPostSaveHookRequestContainer($paymentMethodMapper, $paymentEntity, $quoteTransfer);
+
+        if ($paymentEntity->getPaymentMethod() === PayoneApiConstants::PAYMENT_METHOD_KLARNA) {
+            $this->payoneRequestProductDataMapper->mapData($quoteTransfer, $requestContainer);
```

Comment:
$requestContainer = $this->payoneRequestProductDataMapper->mapData($quoteTransfer, $requestContainer);

Reply:
++